modules/library/main/src/main/java/org/geotools/util/ColorConverterFactory.java

```diff
@@ -43,66 +43,92 @@
  */
 public class ColorConverterFactory implements ConverterFactory {
 
+    /**
+     * Uses {@link Color#decode(String)} to convert String to Color.
+     */
+    public static Converter CONVERT_STRING = new Converter() {
+        public <T> T convert(Object source, Class<T> target) throws Exception {
+            String rgba = (String) source;
+            try {
+                return target.cast(Color.decode(rgba));
+            } catch (NumberFormatException badRGB) {
+                // unavailable
+                return null;
+            }
+        }
+    };
+
+    /**
+     * Converts provided integer to color, taking care to allow rgb and rgba support.
+     */
+    public static Converter CONVERT_NUMBER = new Converter() {
+        public <T> T convert(Object source, Class<T> target) throws Exception {
+            Number number = (Number) source;
+            // is it an integral number, and small enough to be an integer?
+            if (((int) number.doubleValue()) == number.doubleValue()
+                    && number.doubleValue() < Integer.MAX_VALUE) {
+                int rgba = number.intValue();
+                int alpha = 0xff000000 & rgba;
+                return target.cast(new Color(rgba, alpha != 0));
+            } else {
+                return null;
+            }
+        }
+    };
+
+    /**
+     * Converts color to hex representation.
+     */
+    private Converter CONVERT_COLOR_TO_STRING = new Converter() {
+
+        public <T> T convert(Object source, Class<T> target) throws Exception {
+            Color color = (Color) source;
+
+            String redCode = Integer.toHexString(color.getRed());
```

Comment:
This code should check the hint, and look up in the CSS_COLORS map to see if the provided color can be represented as a name.